xrpl/models/transactions/delegate_set.py

```diff
@@ -0,0 +1,149 @@
+"""Model for DelegateSet transaction type."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Dict, List, Optional, Union
+
+from typing_extensions import Self
+
+from xrpl.models.nested_model import NestedModel
+from xrpl.models.required import REQUIRED
+from xrpl.models.transactions.transaction import Transaction
+from xrpl.models.transactions.types import TransactionType
+from xrpl.models.utils import KW_ONLY_DATACLASS, require_kwargs_on_init
+
+PERMISSIONS_MAX_LENGTH = 10
+
+NON_DELEGATABLE_TRANSACTIONS = {
+    TransactionType.ACCOUNT_SET.value,
+    TransactionType.SET_REGULAR_KEY.value,
+    TransactionType.SIGNER_LIST_SET.value,
+    TransactionType.DELEGATE_SET.value,
+    TransactionType.ACCOUNT_DELETE.value,
+}
+
+
+class GranularPermission(str, Enum):
+    """
+    These permissions would support control over some smaller portion of a transaction,
+    rather than being able to do all of the functionality that the transaction allows.
+    """
+
+    TRUSTLINE_AUTHORIZE = "TrustlineAuthorize"
+    """Authorize a trustline."""
+
+    TRUSTLINE_FREEZE = "TrustlineFreeze"
+    """Freeze a trustline."""
+
+    TRUSTLINE_UNFREEZE = "TrustlineUnfreeze"
+    """Unfreeze a trustline."""
+
+    ACCOUNT_DOMAIN_SET = "AccountDomainSet"
+    """Modify the domain of an account."""
+
+    ACCOUNT_EMAIL_HASH_SET = "AccountEmailHashSet"
+    """Modify the EmailHash of an account."""
+
+    ACCOUNT_MESSAGE_KEY_SET = "AccountMessageKeySet"
+    """Modify the MessageKey of an account."""
+
+    ACCOUNT_TRANSFER_RATE_SET = "AccountTransferRateSet"
+    """Modify the transfer rate of an account."""
+
+    ACCOUNT_TICK_SIZE_SET = "AccountTickSizeSet"
+    """Modify the tick size of an account."""
+
+    PAYMENT_MINT = "PaymentMint"
+    """Send a payment for a currency where the sending account is the issuer."""
+
+    PAYMENT_BURN = "PaymentBurn"
+    """Send a payment for a currency where the destination account is the issuer."""
+
+    MPTOKEN_ISSUANCE_LOCK = "MPTokenIssuanceLock"
+    """Use the MPTIssuanceSet transaction to lock (freeze) a holder."""
+
+    MPTOKEN_ISSUANCE_UNLOCK = "MPTokenIssuanceUnlock"
+    """Use the MPTIssuanceSet transaction to unlock (unfreeze) a holder."""
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True, **KW_ONLY_DATACLASS)
+class Permission(NestedModel):
+    """Represents one entry in a Permissions list used in DelegateSet
+    transaction.
+    """
+
+    permission_value: Union[
+        TransactionType,
+        GranularPermission,
+        str,
+    ] = REQUIRED  # type: ignore
```

Comment:
```suggestion
    permission_value: Union[
        TransactionType,
        GranularPermission
    ] = REQUIRED  # type: ignore
```

This would also simplify the logic below - you'd no longer need `_normalize`